huaweicloud/services/eg/data_source_huaweicloud_eg_custom_event_channels.go

```diff
@@ -0,0 +1,171 @@
+)
+
+func DataSourceCustomEventChannels() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCustomEventChannelsRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The region where the custom event channels are located.",
+			},
+			"channel_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The channel ID used to query specified custom event channel.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The channel name used to query specified custom event channel.",
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The ID of the enterprise project to which the custom event channels are belong.",
+			},
+			"channels": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the custom event channel.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the custom event channel.",
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The description of the custom event channel.",
+						},
+						"provider_type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The type of the custom event channel.",
+						},
+						"enterprise_project_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the enterprise project to which the custom event channel belongs.",
+						},
+						"cross_account_ids": {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Description: "The list of domain IDs (other tenants) for the cross-account policy.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"created_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The creation time of the custom event channel.",
+						},
+						"updated_at": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The latest update time of the custom event channel.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func filterCustomEventChannels(channels []custom.Channel, epsId string) ([]interface{}, error) {
+	filter := map[string]interface{}{
+		"EnterpriseProjectId": epsId,
+	}
+
+	filterResult, err := utils.FilterSliceWithField(channels, filter)
+	if err != nil {
+		return nil, fmt.Errorf("error filting list of custom event channels: %s", err)
+	}
+	return filterResult, nil
+}
```

Comment:
How about handle like this. Filtering data does not require throwing errors, and can reduce two type conversions.
```
func filterCustomEventChannels(channels []custom.Channel, epsId string) []custom.Channel {
	if epsId == "" {
		return channels
	}

	var rst []custom.Channel
	for _, v := range channels {
		XXX
	}
	return rst
}
```

Reply:
maybe more filter parameters will append.